### ISSUE TITLE
uBO and ABP behave differently on blob: filters, website breaks with uBO but works with ABP

### ISSUE THREAD

**Issue by Okamoi** [uBlockOrigin/uAssets#2073]
### URL(s) where the issue occurs

The issue can be seen on `vidlox.me` with ListeFR enabled. Random example: `https://vidlox.me/embed-xzp9ka4l2zy5.html`

### Describe the issue

To confirm the issue, I installed uBO on a fresh Firefox profile. Default everything, except I disabled all filter lists, while keeping EasyList and ListeFR enabled. When loading the URL, the video doesn't work.

Doing the same with ABP instead of uBO, the video does work.

### Screenshot(s)

[Screenshot(s) for difficult to describe visual issues are **mandatory**]

### Versions

- Browser/version: Firefox 59.0.2 64-bit, Windows 7
- uBlock Origin version: 1.16.0

### Settings

- Just make sure ListeFR is enabled

### Notes

The cause is ListeFR's rule `|blob:$domain=vidlox.me`. The domain list for this rule is much larger and there's also `vidlox.tv` in there, yet if you replace `.me` with `.tv` in the URL I gave, you'll notice that the video runs fine. Only on `vidlox.me` does the problem occur.

Exception rule `@@|blob:$domain=vidlox.me` fixes the problem.

I reported it to relevant filter list authors and it appears that ABP doesn't break the video, so I've been redirected here. Indeed, if you run a fresh Firefox profile and install ABP with only EasyList+ListeFR, the video runs fine.

So there appears to be a difference in the way uBO and ABP handle `blob:` filters. I wonder which one is more appropriate; just because it breaks on uBO doesn't mean it's the one in the wrong (or in the right for that matter)

### Curiosity questions

- I'm curious to get insight as to how uBO deals with `blob:` rules, it seems that it turns that into some kind of CSP rule that produces console output, but I'm completely unfamiliar with CSP so that didn't help.

- I would also like to understand the purpose of a rule as broad as `|blob:`, isn't it like blocking an entire protocol ? Why can't it be more fine-grained like other types of URLs ?

Thanks

**Comment by uBlock-user** [comment]
Because of this hueg CSP policy being applied in uBO by FRList - `*$csp=default-src 'self' * data: 'unsafe-inline' 'unsafe-eval',domain=10nowvideo.com|11nowvideo.com|123movies.net|12nowvideo.com|1337x.st|1337x.to|13nowvideo.com|14nowvideo.com|1nowvideo.com|2nowvideo.com|3nowvideo.com|5nowvideo.com|6nowvideo.com|7nowvideo.com|8nowvideo.com|9nowvideo.com|bitvid.sx|bmovies.to|bypassed.gold|bypassed.st|cloudtime.to|datpiff.com|divxstage.to|dp-stream.com|dpstream.net|dragon-ball-super-streaming.com|extratorrent.cd|fmovies.me|fmovies.se|freewarefiles.com|fruitadblock.net|fruitstreams.com|immunicity.gold|immunicity.st|immunicity.world|istreamin.com|japscan.cc|japscan.com|keepvid.com|kinogo.club|kinotochka.net|kshowonline.com|lavamovies.com|likeupload.org|lucariostream.eu|movshare.ag|movshare.net|novamov.ag|novamov.com|nowdownload.ag|nowdownload.at|nowdownload.co|nowdownload.ec|nowdownload.eu|nowdownload.sc|nowdownload.to|nowvideo.ag|nowvideo.co|nowvideo.cz|nowvideo.ec|nowvideo.li|nowvideo.sx|nowvideo.to|oload.info|oload.stream|oload.tv|primewire.ag|proxyportal.me|putlocker.io|seasonvar.ru|sokrostream.biz|sokrostream.cc|sokrostream.cx|sokrostream.tv|sokrostream.ws|streamango.com|streamdown.cc|streamdown.tv|streamgaroo.com|streaming-boruto.com|streaming-dragon-ball-super.com|thecountrycaller.com|unblockall.org|unblocked.bid|unblocked.gold|unblocked.lol|unblocked.st|unlockproj.cricket|videoweed.com|videoweed.es|videowood.tv|vidgg.to|vidlox.me|vidlox.tv|vidwatch3.me|watchvideo.us|watchvideo11.us|watchvideo12.us|watchvideo13.us|watchvideo14.us|watchvideo15.us|watchvideo5.us|x1337x.eu|x1337x.ws`

**Comment by mapx-** [comment]
@gorhill  did you add the french list ?

**Comment by Okamoi** [comment]
> What is that filter supposed to accomplish?

I don't know, it's not explained in the list, maybe they have an online commit thing that gives insight though.

At any rate, it is listed under the list's "Third-party ads" category and along with a set of rules that have very long domain lists and block `websocket`, `webrtc` and `blob:`. All three rules seem to apply in very large majority to video-related domains. (Bit of torrents for `websocket`)

**Comment by mapx-** [comment]
mmhmm ...right, I get no blob: filter in ABP

**Comment by okiehsch** [comment]
If the blob filter does not accomplish anything with Adblock Plus, would it not be best to mark it as badfilter, as I am not sure what exactly it doesaccomplish if one uses uBO, other then breaking video functionality on a number of sites.
Right now we need a blob exception filter for
````
dailycaller.com
investopedia.com
nydailynews.com
tomsguide.com
tomshardware.com
twitch.tv
vidlox.me
vidlox.tv
````
and this are only the known issues, there are probably more sites broken because of this.

**Comment by Okamoi** [comment]
It appears that @SMed79 [made that filter](https://hg.adblockplus.org/listefr/rev/0abb6892f7bb) with stated reason "popup".

He is also [the person who directed me](https://forums.lanik.us/viewtopic.php?f=91&t=40570) there after showing me a UI video where you can see that ABP doesn't have the issue. That's what led me here.

Based on the video he made, it seems that the creator of the blob: rule is an ABP user, which is odd if ABP ignores `blob:` rules.

**Comment by SMed79** [comment]
@Okamoi I directed you here to fix a possible bug in **uBO** but no one explain why **ABP** is not affected ?!

@gorhill Do you think removing the filter is the best fix?

**Comment by Okamoi** [comment]
> I directed you here to fix a possible bug in uBO but no one explain why ABP is not affected ?!

Just adding a little noise to tell you I'm not blaming anything, just trying to tie the rule with a person able to explain it to Gorhill :)

I'm super thankful for your work and have been so for years.

If `blob:` ads are a thing it's really annoying since as I understand it, making such rules more fine-grained should be problematic from a performance viewpoint. Still, I'd rather have degraded performance than ads or trackers.

**Comment by gorhill** [comment]
> should be problematic from a performance viewpoint

Using a `csp=` to enforce blocking means uBO merely injects the CSP and the browser will do all the blocking with uBO having nothing to do at all. Performance concerns for `|blob:`-like filters are a non-issue.

**Comment by Okamoi** [comment]
If they block the entire `blob:` "protocol" or whatever it is, yeah, but isn't that a little too heavy handed ? Can't a site have both legitimate `blob:` and crap `blob:` ? In which case, parsing becomes necessary ?

Well this is a little out of scope for this issue, but I wanted to leave a trace since removing the filter altogether has been considered

**Comment by okiehsch** [comment]
````
|blob:$domain=101greatgoals.com|1337x.to|1channel.biz|4chan.org|allthetests.com|ancient-origins.net|androidcentral.com|antonymsfor.com|auroravid.to|biology-online.org|bitvid.sx|breakingisraelnews.com|britannica.com|btdb.in|btdb.to|champion.gg|closerweekly.com|cloudtime.to|colourlovers.com|connectedly.com|convert-me.com|convertcase.net|couch-tuner.at|couch-tuner.me|couchtuner.ac|couchtuner.us|crackberry.com|dailycaller.com|datpiff.com|demonoid.pw|destructoid.com|dietlast.com|dreamfilm.se|episodetube.com|episodetube.net|ewatchseries.to|fastpic.ru|filme-streamz.com|filmlinks4u.is|firstforwomen.com|firstrowau.eu|firstrowus1.eu|fmovies.is|fmovies.se|fmovies.to|fullmatchesandshows.com|gamesradar.com|getinmybelly.com|gofirstrow.eu|gorillavid.in|hdfilme.tv|homerun.re|igvnews.co.uk|imagefap.com|imgadult.com|imgtaxi.com|imgwallet.com|imore.com|investopedia.com|israelnationalnews.com|jerusalemonline.com|jewsnews.co.il|keepvid.com|kickass.cd|kino-streamz.com|kiplinger.com|kissmanga.com|kshowonline.com|letmewatchthis.pl|letmewatchthis.video|livecricketz.org|livescience.com|lolcounter.com|ludokado.com|merriam-webster.com|movies4stream.com|movpod.in|mylivecricket.org|mywatchseries.to|newtvworld.com|noslocker.com|nosvideo.com|nowfeed2all.eu|nowvideo.li|nowvideo.sx|nowvideo.to|nydailynews.com|olympicstreams.me|omgcheckitout.com|onwatchseries.to|openload.co|pcgamer.com|pcgames-download.net|phonearena.com|phonesreview.co.uk|pocketnow.com|primewire.is|primewire.to|primewire.unblockall.xyz|psu.com|rinf.com|roadracerunner.com|serietvsubita.net|sgvideos.net|skidrowcrack.com|snowysmile.com|sockshare.net|sportspickle.com|streamazone.com|streamgaroo.com|strikeout.co|strikeout.me|strikeout.mobi|teamliquid.net|tetrisfriends.com|textsfromlastnight.com|thefreethoughtproject.com|thevideo.me|thewatchseries.ac|timesofisrael.com|tomsguide.com|tomshardware.co.uk|tomshardware.com|torrentz2.eu|torrentz2.me|trifind.com|tv-series.me|unblckd.org|veteranstoday.com|videotoolbox.com|vidlox.tv|vidtodo.com|vidup.me|vidwatch3.me|vipbox.bz|vipbox.is|vipbox.nu|vipbox.sx|vipbox.tv|vipboxeu.co|vipboxoc.co|vipboxtv.me|vipleague.ch|vipleague.co|vipleague.is|vipleague.me|vipleague.mobi|vipleague.se|vipleague.sx|vipleague.tv|vipleague.ws|vipstand.is|vivo.sx|vrheads.com|watchcartoononline.io|watchepisodes-tv.com|watchseries.sk|watchvideo.us|watchvideo10.us|watchvideo11.us|watchvideo12.us|watchvideo13.us|watchvideo14.us|watchvideo15.us|webfirstrow.eu|wholecloud.net|windowscentral.com|world4ufree.ws
````

is the filter in `EasyList`, now looking at the domains I am pretty sure many were added because they use uponit or pagefair ad-reinsertion scripts, not because of popups. 
The best example is `nydailynews.com`, which uses pagefair but does certainly not use `blob` to circumvent popup-blocking.

**Comment by ryanbr** [comment]
blob may've been used in the past, but sites do change

**Comment by okiehsch** [comment]
I checked all non streaming sites and the videos on
````
closerweekly.com
firstforwomen.com
livescience.com
pcgamer.com
````
are also broken.

**Comment by ryanbr** [comment]
So how is it broken? I visited pcgamer.com, it looked fine?

**Comment by okiehsch** [comment]
The youtube videos work, but their own hosted videos are broken.
Examplle link:
`https://www.pcgamer.com/battletech-review/`
scroll down to the recommended videos.
![unbenannt](https://user-images.githubusercontent.com/16838044/39226619-d0b263f2-4853-11e8-8fdf-5f838effb4f4.JPG)

**Comment by mapx-** [comment]
https://github.com/easylist/easylist/commit/f494726f1fded77060544faf9144fea5e8a083d5
https://github.com/uBlockOrigin/uAssets/commit/c17c3670d6d3a8aa7de5186b52a45e2d20a7d65b
